src/libstore/derivations.cc

```diff
@@ -518,6 +518,27 @@ DerivationOptions DerivationOptions::fromEnv(const StringPairs & env)
 
     DerivationOptions defaults = {};
 
+    if (auto structuredAttrs = parsed.getStructuredAttrs()) {
```

Comment:
Let me know if you think this should rather stay [here](https://github.com/haenoe/nix/blob/6f64154eeac328c87e62c1d8ae3aacd90ab9d5d7/src/libstore/unix/build/local-derivation-goal.cc#L2906) (in `local-derivation-goal.cc`).

Reply:
Reverted this, because the change pollutes the output (as we parse with `DerivationOptions::fromEnv` multiple times while evaluating a derivation.